extensions/amp-ad-exit/0.1/amp-ad-exit.js

```diff
@@ -190,6 +204,74 @@ export class AmpAdExit extends AMP.BaseElement {
       this.user().error(TAG, 'Invalid JSON config', e);
       throw e;
     }
+
+    this.ampAdResourceId_ = user().assert(
+        this.getAmpAdResourceId_(),
+        `${TAG}: No friendly parent amp-ad element was found for amp-ad-exit.`);
+
+    this.unlisten_ = listen(this.getAmpDoc().win, 'message', event => {
+      const responseMessage = deserializeMessage(getData(event));
+      this.assertValidResponseMessage_(responseMessage, event.origin);
```

Comment:
I'm not sure I understand? I think if one event is received, and we validate that it comes from the expected origin, and then another event is received, I think we should still validate that event #2 comes from the expected origin. We could locally cache [vendor, origin], but then we'd just be trading the map lookup in assertVendor() for a local map lookup. But maybe there is an optimization I am not seeing?

Reply:
With the number of vendors increasing, `assertVendor()` can become more expensive I think. 
What's the possibility that vendor A with originA pretend to have vendor nameB?
I am think of something like
```
const verifiedOrigin;
if (!verifedOrigin['originA']) {
  if (vendorA['origin'] == 'originA') { 
    verifedOrigin['originA'] = true;
  }
}
```
so once an origin is verified, we don't need to verify it again? 

Or I am also thinking something like verify  `event.source`. Once the origin is verified, we trust message from the source. don't know if that's doable though😄

Reply:
> "What's the possibility that vendor A with originA pretend to have vendor nameB?"
Since only accredited vendors can be involved here, I don't think anyone would actually do this. But from a technical standpoint, it would not be difficult.
@keithwrightbos Thoughts?